spec/order_spec.rb

```diff
@@ -0,0 +1,53 @@
+require_relative '../lib/order'
+
+describe Order do
+  
+  subject(:order) { described_class.new } # allows the use of 'dish' instead of 'subject' for clarity  
+  
+  describe "#show_menu" do
+    it 'shows the dishes on the menu' do
+      expect { order.show_menu }.to output("Sweetcorn Patties 3.99\nThai Red Curry (King Prawn) 6.99\nThai Red Curry (Vegetable) 5.99\nSingapore Noodles (Vegetable) 4.99\nSteamed Rice 2.99\n").to_stdout
+    end
+  end
+
+  describe "#open_order" do
+    it 'opens the order with the menu item quantities all intialised to 0' do
+      order.show_menu
+      order.open_order
+      expect(order.order_items.length).to eq 5
+    end
+  end
+
+  describe "#add_dish" do
+    it 'this increases the quantity of the dish by the given amount, or 1 by default' do
+      order.show_menu
+      order.open_order
+      order.add_dish("Sweetcorn Patties", 2)
+      expect(order.order_items[:"Sweetcorn Patties"]).to eq 2 # rubocop doesn't like "Swe..".to_sym
+      order.add_dish("Singapore Noodles (Vegetable)")
+      expect(order.order_items[:"Singapore Noodles (Vegetable)"]).to eq 1
+    end
+  end
+
+  describe "#check_order" do
+    it 'check that the order items, amounts and overall price is what the customer is expecting' do
+      order.show_menu
+      order.open_order
+      order.add_dish("Sweetcorn Patties", 2)
+      order.add_dish("Singapore Noodles (Vegetable)")
+      order.check_order
+      expect(order.total).to eq 12.97
+    end
+    it 'check that the order items, amounts and overall price is what the customer is expecting' do
+      order.show_menu
+      order.open_order
+      order.add_dish("Sweetcorn Patties", 2)
+      order.add_dish("Singapore Noodles (Vegetable)")
+      # order.check_order
+      expect { order.check_order }.to output("You have ordered:\nSweetcorn Patties x2 at £3.99 each\nSingapore Noodles (Vegetable) x1 at £4.99 each\nThe total cost of your order is £12.97\n").to_stdout
+   
+      expect(order.total).to eq 12.97
+    end
+  end
+
+end
```

Comment:
These blocks are functional, but they're a wee bit long. For instance, I'm not sure that we need to 'show menu' each time - I think we can assume that if the user wants to check their order that they won't need to see the menu again :) 
I think that you could have removed your 'open_order' method too. You've already initialized your '@order_items' hash within your 'order.rb' file, so instead of calling the 'open_order' method each time, you could have just had your 'add_dish' method push the dish that was being selected into your '@order_items' hash!